core/src/smartcontracts/wasm.rs

```diff
@@ -1460,132 +1515,142 @@ impl<S> RuntimeBuilder<S> {
 macro_rules! create_imports {
     (
         $linker:ident,
-        $(export::$name:ident => $fn_path:path),* $(,)?
+        $ty:ty,
+        $(export::$name:ident => $fn:expr),* $(,)?
     ) => {
         $linker.func_wrap(
                 WASM_MODULE,
                 export::LOG,
-                Runtime::log,
+                |caller: ::wasmtime::Caller<$ty>, offset, len| Runtime::log(caller, offset, len),
```

Comment:
It's because mutable references are invariant.
Previously there were only immutable references (they are covariant in lifetime). 
I don't remember this quite well since it was few month ago.